src/run.jl

```diff
@@ -180,144 +206,111 @@ function run_sandboxed_test(install::String, pkg; log_limit = 2^20 #= 1 MB =#,
             println()
         finally
             print("\n\n", '#'^80, "\n# PkgEval teardown: $(now())\n#\n\n")
-
-            # give the host some time to pick the above marker up, and kill us
-            sleep(60)
         end"""
     cmd = do_depwarns ? `--depwarn=error` : ``
     cmd = `$cmd --eval 'eval(Meta.parse(read(stdin,String)))' $(pkg.name)`
 
     input = Pipe()
     output = Pipe()
 
-    function stop()
-        kill_container(container)
+    env = Dict(
+        "JULIA_PKG_PRECOMPILE_AUTO" => "0",
+        # package hacks
+        "PYTHON" => "",
+        "R_HOME" => "*"
+    )
+    if haskey(ENV, "JULIA_PKG_SERVER")
+        env["JULIA_PKG_SERVER"] = ENV["JULIA_PKG_SERVER"]
     end
 
-    container_lock = ReentrantLock()
-
-    p = run_sandboxed_julia(install, cmd; name=container, tty=false, wait=false,
-                                          stdout=output, stderr=output, stdin=input,
-                                          kwargs...)
+    proc = run_sandboxed_julia(install, cmd; env, wait=false,
+                               stdout=output, stderr=output, stdin=input,
+                               kwargs...)
     close(output.in)
 
     # pass the script over standard input to avoid exceeding max command line size,
     # and keep the process listing somewhat clean
     println(input, script)
     close(input)
 
+    function stop()
+        if process_running(proc)
+            # FIXME: if we only kill proc, we sometimes only end up killing the sandbox.
```

Comment:
This happens even with SIGINT, which `sandbox.c` probably should handle (i.e. forward to its child).